### ISSUE TITLE
[OJS] Add reviews to email button selects the wrong option for radio buttons in some cases

### ISSUE THREAD

**Issue by ajnyga** [pkp/pkp-lib#2437]
Scenario:
A new custom review form is created and it includes a selection with radio buttons. The editor first adds these options and saves the question:
```
Option 1
Option 2
Option 3
```
After a while the editor adds some new values and removes some of the old one:
```
Option 3
Option 4
Option 5
```

The form is saved and used in reviews. The radio button selection works fine in the form and the correct alternative is also shown when the editor reads the incoming reviews. **However**, when the editor clicks the `Add reviews to email` button a wrong selection is shown.

This is because in the case of radiobuttons the value in `review_form_element_settings / possibleResponses` now has an array that does not start from 0. However, when the radiobuttons are shown in a form they get a running number as the value, and this same value is saved in the `review_form_responses` table leading in to a mismatch when this is called: https://github.com/pkp/pkp-lib/blob/ojs-stable-3_0_2/classes/controllers/modals/editorDecision/PKPEditorDecisionHandler.inc.php#L255

**NOTE: does this apply to checkboxes as well?**

I guess three alternatives here: 
- resequence the array when creating the form if some values are removed. 
- Or add a custom way of handling things for radio button here https://github.com/pkp/pkp-lib/blob/ojs-stable-3_0_2/classes/controllers/modals/editorDecision/PKPEditorDecisionHandler.inc.php#L255
- or use the actual key of the array in the radio button value and always save that value to the database as the selected option

**Comment by ajnyga** [comment]
I did a quick fix because one of journals is having trouble with this. I did this, but this is definitely not the final solution here:
```
else if ($reviewFormElement->getElementType() == REVIEW_FORM_ELEMENT_TYPE_RADIO_BUTTONS) {
									ksort($possibleResponses);
									$possibleResponses = array_values( $possibleResponses );
									$body .= '<blockquote>' . PKPString::stripUnsafeHtml($possibleResponses[$reviewFormResponse->getValue()]) . '</blockquote>';
								}
```

**Comment by ajnyga** [comment]
I can think of some solutions here, but the solution would have to take into account the fact that there are several ongoing review processes with forms that have that problem in the array keys.

**Comment by alexxxmendonca** [comment]
We identified the exact same issue in our install.

**Comment by asmecher** [comment]
@ajnyga, yuck -- thanks for tracking this down (and @alexxxmendonca for confirming). I've confirmed that this applies to both checkboxes and radio buttons, and applied a fix like your proposal. This is something that can be cherry-picked or patched on an OJS 3.0.2 installation without requiring any database changes. (Apply https://github.com/pkp/pkp-lib/commit/8ec91ecd0b3abe0d92a14c7fdeae3bf36562a85c in `lib/pkp`.)

Reviewing OJS 2.x vs 3.x: OJS2 used array offsets for radio buttons, checkboxes, _and_ dropdowns. OJS 3.x only uses array offsets for checkboxes and radio buttons, and uses array indexes for dropdowns. Fortunately OJS 2.x regenerates array indexes every time review forms are edited, so the two types of addressing should be equivalent, meaning that content migrated from OJS 2.x should all be OK.

As you've noted, @ajnyga, this is probably not the "right" fix. I've filed that separately in https://github.com/pkp/pkp-lib/issues/2477 for later work. This fix will make everything accurate and fix the `ojs-stable-3_0_2` branch without database intervention, and with that resolved we can improve storage at our convenience.

This tests fine for me, but either of your confirmation would be welcome!
